### PR TITLE
New test runner

### PR DESCRIPTION
**Pull request by lpil**
To get us to runner specification version 2. https://exercism.org/docs/building/tooling/test-runners/interface

- [x] New runner made
- [ ] Published to Hex
- [ ] Packages directory updated to use it
- [ ] Scripts updated to use it
- [ ] Exercises updated to use it (post-merge)

### CODE REVIEW

**Comment by iHiD** [comment]
Whoop whoop!

**Review by JohnBjrk — APPROVED** [review]
Seems like a solid first version to me! As noted in the comments there could be some more checks around accessing the modules/functions, but given that this will run in the specific context of an exercise I don't think it's a problem.

**Comment by lpil** [comment]
@exercism/gleam let me know your Hex logins and I will add you to the Hex repo so you have push rights

File: runner/src/exercism/test_runner.gleam [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  let assert Ok(_) = case erlang.start_arguments() {
+    ["--json-output-path=" <> path] | ["--json-output-path", path] -> {
+      let json = internal.results_to_json(results)
+      simplifile.write(json, path)
+    }
+    _ -> Ok(Nil)
+  }
+
+  halt(case passed {
```
Comment (michallepicki) [inline runner/src/exercism/test_runner.gleam:30]: Is this correct? From the test runner specification:
> The runner must exit with an exit code of 0 if it has run successfully, regardless of the status of the tests.

Comment (lpil) [reply runner/src/exercism/test_runner.gleam:30]: Aye, there's a script that runs it and handles things like the exit status and compile errors

https://github.com/exercism/gleam-test-runner/blob/main/bin/run.sh

**Pull request merged**